apis/v1/secretproviderclass_types.go

```diff
@@ -46,13 +46,32 @@ type SecretObject struct {
 	Data        []*SecretObjectData `json:"data,omitempty"`
 }
 
+type Secret struct {
+	// name of the object to sync
+	ObjectName string `json:"objectName,omitempty"`
+	// expected format of the secret received from the provider (e.g. plaintext, json)
+	Format string `json:"format,omitempty"`
+	// JSON path to target for a secret received in JSON format
+	JsonPath string `json:"jsonPath,omitempty"`
+}
+
+type TransformOptions struct {
+	// expected format of the secret received from the provider (e.g. plaintext, json)
+	Format string `json:"format,omitempty"`
+	// JSON path to target for a secret received in JSON format
```

Comment:
I notice from the implementation that it expects the `JsonPath` to resolve to string key's that become filenames. I think this comment should be explicit about that behavior.

Additionally I think it might be nice to support extracting a single value if the path resolves to a string instead of a json object. It may be a bit more verbose but I think more explicit to support resolving to a json value and adding an `OutFilePath` parameter. Consider a response from the provider that says file `my_secret` has value `{"username":"admin", "password": "hunter2"}`

You could then write an SPC like:

```
apiVersion: secrets-store.csi.x-k8s.io/v1
kind: SecretProviderClass
metadata:
  name: app-secrets
spec:
  provider: vault
  parameters:
    objects: |
      - objectName: "my_secret"
        secretPath: "secret/data/my-kv"
        secretKey: "blob"
  transforms:
    - inFilePath: "my_secret"
      jsonPath: "$.username"
      outFilePath: "username"
    - inFilePath: "my_secret"
      jsonPath: "$.password"
      outFilePath: "password"
```

Later we could add a transform like the this to resolve https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/948

```
  transforms:
    - staticValue: "admin"
      outFilePath: "username"
```

Where we could generate a file based just on SPC data for non-secret static value `admin` to file `username`.